drivers/dma/dma_stm32f4x.c

```diff
@@ -341,19 +341,23 @@ static int dma_stm32_config_devcpy(struct device *dev, u32_t id,
 static int dma_stm32_config_memcpy(struct device *dev, u32_t id)
 {
 	struct dma_stm32_device *ddata = dev->driver_data;
-	struct dma_stm32_stream_reg *regs = &ddata->stream[id].regs;
+	if (ddata->mem2mem) {
```

Comment:
Using mem2mem is good, but it seems it is initialized only for instance 2.
Can you add its configuration to false for instance 1?